docs/architecture/_index.md

```diff
@@ -0,0 +1,188 @@
+imageAlt: "Viam architecture"
+images: ["/get-started/installation/thumbnails/configure.svg"]
+tags: ["components", "services", "communication"]
+menuindent: true
+---
+
+![Viam data, ML and registry in the cloud and Viam fleet and build on-premises. Viam core is the communication layer that connects the cloud to the on-device portion.](/architecture/cloud-premises.png)
+
+TODO: replace the above diagram with something similar but that more clearly shows viam-server etc.
+
+## Basic structure of a simple machine
+
+<p>
+{{<imgproc src="/architecture/simple-machine.png" class="alignright" resize="x1100" declaredimensions=true alt="viam-server running on a board connected to a sensor. Data is stored on a local folder and synced to a folder in the Viam app cloud." style="max-width:400px" >}}
+</p>
+
+Imagine you have a simple device consisting of a temperature sensor connected to the GPIO pins of a single-board computer (SBC).
+You want data from the sensor to be captured at regular intervals, and synced to the cloud.
+To set this up, you would do the following:
+
+- Install `viam-server` on the SBC.
+- Create a configuration file in the Viam app indicating that you are using a sensor {{< glossary_tooltip term_id="component" text="component" >}}, and indicating which pins of the SBC it is connected to.
+- Configure data capture and sync:
+  - Edit your configuration to indicate the interval at which to capture data, and the interval at which to sync it.
+
+When `viam-server` starts, it uses credentials stored locally to establish a connection with the Viam app over {{< glossary_tooltip term_id="webrtc" text="WebRTC" >}}.
+It fetches its configuration, which in our example contains the sensor pin information, from the Viam app.
+
+<details>
+  <summary>Click for information on local configuration</summary>
+  <p>
+  If you need to run your machine offline, you can manually create a local configuration file, eliminating the need for <code>viam-server</code> to fetch its config.<br><br>
+  The advantages of using the credentials and pulling the config from the Viam app are:
+  <ul>
+    <li>The config builder UI in the Viam app is more user-friendly than writing JSON manually</li>
+    <li>You can easily share configs across multiple identical machines and update them from one place</li>
+    <li>etc....TODO</li>
+  </ul>
+  If your machine usually runs offline but can connect online occasionally, it can pull a configuration from the Viam app and cache it locally to use until the next time it has an internet connection.
+  </p>
+</details><br>
+
+`viam-server` runs your machine, securely managing all communications between hardware and to the cloud.
+
+You can use the tools in the Viam app to remotely view sensor data as well as to change your machine's configuration, to view logs, and more.
+
+Now imagine you want to run code to send you an email when the temperature sensor reads over 100 degrees Fahrenheit:
```

Comment:
This isn't a great example because you'd actually not want to run code, you'd want to configure a Trigger that does this. However I can't think of a better example either right now.

Reply:
Good point

Reply:
I mean it could be same sensor example, but you want to run code to spin up a custom dashboard instead of send you a notification?